types/material-ui/index.d.ts

```diff
@@ -869,7 +880,7 @@ declare namespace __MaterialUI {
     namespace DatePicker {
         interface DatePickerProps {
             // <TextField/> is the element that get the 'other' properties
-            DateTimeFormat?: Intl.DateTimeFormat;
+            DateTimeFormat?: typeof Intl.DateTimeFormat;
```

Comment:
No, because it can be polyfilled and the constructor definition should match exactly as material-ui makes extensive use of DateTimeFormat's standard. a simple `new () => ...` will not cover all cases. But the typings provided by typescript does.